fhir-server-test/src/test/java/com/ibm/fhir/server/test/terminology/CodeSystemClearCacheOperationTest.java

```diff
@@ -0,0 +1,104 @@
+    public static final boolean DEBUG = false;
+
+    // URLs to call against the instance
+    public static final String BASE_VALID_URL = "/CodeSystem";
+    
+    public static final String CODE_SYSTEM_ID = "test";
+    public static final String CODE_SYSTEM_URL = "http://ibm.com/fhir/CodeSystem/test";
+    public static final String CODE_SYSTEM_VERSION = "1.0.0";
+
+    @BeforeClass
+    public void setup() throws Exception {
+        Response response = doPut("CodeSystem", CODE_SYSTEM_ID, "testdata/CodeSystem-test.json");
+        assertEquals(response.getStatusInfo().getFamily(), Response.Status.Family.SUCCESSFUL);
+    }
+    
+    @Test(groups = { TEST_GROUP_NAME })
+    public void testClearCacheParameterVariations() throws Exception {
+        Response response;
+        
+        response = doGet(BASE_VALID_URL + "/$clear-cache");
+        assertEquals(response.getStatus(), Response.Status.OK.getStatusCode());
+        
+        response = doGet(BASE_VALID_URL + "/$clear-cache", "url", CODE_SYSTEM_URL);
+        assertEquals(response.getStatus(), Response.Status.OK.getStatusCode());
+        
+        response = doGet(BASE_VALID_URL + "/$clear-cache", "url", CODE_SYSTEM_URL, "codeSystemVersion", CODE_SYSTEM_VERSION);
+        assertEquals(response.getStatus(), Response.Status.OK.getStatusCode());
+        
+        clearCache();
+    }
+    
+    @Test(groups = { TEST_GROUP_NAME })
+    public void testClearCacheValidateCode() throws Exception {
+        Response response;
+        Parameters parameters;
+        
+        // Make sure the resource is there
+        response = doPut("CodeSystem", CODE_SYSTEM_ID, "testdata/CodeSystem-test.json");
+        assertEquals(response.getStatusInfo().getFamily(), Response.Status.Family.SUCCESSFUL);
+        
+        // Once to make sure it is gone from the cache
+        clearCache();
+        
+        // Twice to see what happens when it is empty
+        clearCache();
+        
+        // Subsumes to reload the cache
+        parameters = validateCode(SNOMED_CT, "K");
+        assertEquals(getBooleanParameterValue(parameters, "result"), Boolean.TRUE);
+        
+        // Update the resource
+        response = doPut("CodeSystem", CODE_SYSTEM_ID, "testdata/CodeSystem-test-updated.json");
+        assertEquals(response.getStatus(), Response.Status.OK.getStatusCode());
+        
+        parameters = validateCode(SNOMED_CT, "K");
+        assertEquals(getBooleanParameterValue(parameters, "result"), Boolean.TRUE);;
+
+        // After cache is cleared, subsumes outcome should be updated
+        clearCache();
+        
+        parameters = validateCode(SNOMED_CT, "K");
+        assertEquals(getBooleanParameterValue(parameters, "result"), Boolean.FALSE);
+    }
+	
+	private Parameters validateCode(String system, String code) throws Exception {
+		Response response = doGet(BASE_VALID_URL + "/" + CODE_SYSTEM_ID + "/$validate-code", "system", system, "code", code);

```

Comment:
I see several horizontal spacing issues. In some cases it looks like more than 4 spaces are present. In other cases it looks like less than 4 spaces are present. I also what looks to be tab characters. Will need to pull down the raw file to verify.

Reply:
Yeah, sorry about that. My Eclipse had some config issues. I'll do a sweep through again and make sure I've got the styles correct.